src/main/java/io/opentracing/contrib/global/GlobalTracer.java

```diff
@@ -0,0 +1,115 @@
+ * @see Tracer
+ */
+public final class GlobalTracer {
+    private static final Logger LOGGER = Logger.getLogger(GlobalTracer.class.getName());
+    private static final Callable<Tracer> DEFAULT_PROVIDER = new Callable<Tracer>() {
+        public Tracer call() throws Exception {
+            return NoopTracerFactory.create();
+        }
+    };
+
+    /**
+     * The resolved {@link Tracer} to delegate the global tracing implementation to.<br>
+     * This can be either an {@link #register(Tracer) explicitly registered delegate}
+     * or the automatically resolved Tracer implementation.<br>
+     * Management of this reference is the responsibility of the {@link #tracer()} method.
+     */
+    private static final AtomicReference<Tracer> DELEGATE = new AtomicReference<Tracer>();
+
+    /**
+     * Private constructor to prevent instantiation of this utility class.
+     */
+    private GlobalTracer() {
+        throw new UnsupportedOperationException();
+    }
+
+    /**
+     * This method allows explicit registration of a configured {@link Tracer} implementation to back the behaviour
+     * of the {@link #tracer() global tracer} instance.
+     *
+     * @param delegate The delegate tracer to delegate the global tracing implementation to.
+     * @return The previous global tracer, or <code>null</code> if no global tracer was (auto-)registered yet.
+     */
+    public static Tracer register(final Tracer delegate) {
+        final Tracer previous = DELEGATE.getAndSet(GlobalSpanTracer.wrap(delegate));
+        LOGGER.log(Level.INFO, delegate == null ? "Cleared GlobalTracer delegate registration."
+                : "Registered GlobalTracer delegate: {0}.", delegate);
+        return previous;
+    }
+
+    /**
+     * This method returns the {@link #register(Tracer) explicitly registered} Tracer implementation,
+     * or attempts to lazily load an available implementation according to the standard Java SPI conventions.
+     * <p>
+     * If no delegate is found, the {@link io.opentracing.NoopTracer NoopTracer} will be returned and no
+     * {@link GlobalSpanManager#activeSpan() globally-active spans} will be created.
+     *
+     * @return The non-<code>null</code> global tracer to use.
+     */
+    public static Tracer tracer() {
+        Tracer instance = DELEGATE.get();
+        if (instance == null) {
+            final Tracer singleton = GlobalSpanTracer.wrap(ServiceLoader.loadSingleton(Tracer.class, DEFAULT_PROVIDER));
+            while (instance == null && singleton != null) {
+                DELEGATE.compareAndSet(null, singleton);
+                instance = DELEGATE.get();
+            }
+            LOGGER.log(Level.INFO, "Using global Tracer implementation: {0}.", instance);
+        }
+        LOGGER.log(Level.FINEST, "Global tracer: {0}.", instance);
+        return instance;
+    }
+
+    /**
+     * Private wrapper class that delegates all tracing to the specified implementation but makes sure to register
```

Comment:
again.. pretend there are only NN words.. which can we leave out without losing any signal?